queue.c

```diff
@@ -261,6 +261,36 @@ static void prvInitialiseNewQueue( const UBaseType_t uxQueueLength,
         }                                                  \
     }                                                      \
     taskEXIT_CRITICAL()
+
+/*
+ * Macro to increment cTxLock member of the queue data structure. It is
+ * capped at the number of tasks in the system as we cannot unblock more
+ * tasks than the number of tasks in the system.
+ */
+#define prvIncrementQueueTxLock( pxQueue, cTxLock )                   \
+    {                                                                 \
+        const UBaseType_t uxNumberOfTasks = uxTaskGetNumberOfTasks(); \
+        if( ( UBaseType_t ) ( cTxLock ) < uxNumberOfTasks )           \
+        {                                                             \
+            configASSERT( ( cTxLock ) != queueINT8_MAX );             \
+            ( pxQueue )->cTxLock = ( int8_t ) ( ( cTxLock ) + 1 );    \

```

Comment:
Might be better to put the cast on the "1", rather than the result of adding a uint_8 with an integer.